### PR TITLE
[stable/pomerium] Add prometheus operator ServiceMonitor support

### PR DESCRIPTION
**Pull request by travisgroth**
#### What this PR does / why we need it:

Adds/updates templates to support prometheus operator ServiceMonitor configuration

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/pomerium/README.md [hunk 1]
```diff
@@ -75,6 +75,9 @@ Parameter                         | Description
 `service.annotations`             | Service annotations                                                                                                                                                                                        | `{}`
 `service.externalPort`            | Pomerium's port                                                                                                                                                                                            | `443`
 `service.type`                    | Service type (ClusterIP, NodePort or LoadBalancer)                                                                                                                                                         | `ClusterIP`
+`serviceMonitor.enabled`          | Create Prometheus Operator ServiceMonitor                  | `false`
```
Comment (victornoel) [inline stable/pomerium/README.md]: it's more about cosmetic than semantics, but on all the other lines, the example cell is on its own line :)

File: stable/pomerium/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{ if .Values.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "pomerium.fullname" . }}
+{{ if .Values.serviceMonitor.namespace -}}
+  namespace: {{ .Values.serviceMonitor.namespace }}
```
Comment (victornoel) [inline stable/pomerium/templates/servicemonitor.yaml:7]: is it standard practice to specify a different namespace than the one of the monitored service?

File: stable/pomerium/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{{- end }}
+  labels:
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
```
Comment (victornoel) [inline stable/pomerium/templates/servicemonitor.yaml:12]: Shouldn't there be also a `app.kubernetes.io/name: {{ template "pomerium.name" . }}` here (and the same for the selector)? That's what was done for the `Deployment`s at least

File: stable/pomerium/values.yaml [hunk 4]
```diff
@@ -121,3 +121,9 @@ image:
 metrics:
   enabled: false
   port: 9090
+
+serviceMonitor:
+  enabled: false
+  namespace: false
```
Comment (victornoel) [inline stable/pomerium/values.yaml]: if you set this to false, helm will be confused when the user then set it to a string. Either leave it commented or maybe set it to the empty string? (I don't like that second option but I'm not 100% sure what's best).

**Review by victornoel — COMMENTED** [review]
@travisgroth I've left you some comments :)

File: stable/pomerium/templates/servicemonitor.yaml (hunk 2 above)
Comment (travisgroth) [reply stable/pomerium/templates/servicemonitor.yaml:7]: Yes the operator can run in a mode where only specific namespace(s) are watched.  I don't think I've seen it used, but we should be able to support it.

File: stable/pomerium/templates/servicemonitor.yaml (hunk 3 above)
Comment (travisgroth) [reply stable/pomerium/templates/servicemonitor.yaml:12]: It seemed repetitive when rendered out.  Harmless to retain, but it doesn't do much.

The deployments have `app.kubernetes.io/name` with the individual service names, so I left the selector intentionally loose to capture all of the deployments the chart installs with one rule.

File: stable/pomerium/values.yaml (hunk 4 above)
Comment (travisgroth) [reply stable/pomerium/values.yaml]: Good catch 👍 empty string is the right way, I think.

File: stable/pomerium/templates/servicemonitor.yaml (hunk 2 above)
Comment (victornoel) [reply stable/pomerium/templates/servicemonitor.yaml:7]: I see, thanks for the explanation

File: stable/pomerium/templates/servicemonitor.yaml (hunk 3 above)
Comment (victornoel) [reply stable/pomerium/templates/servicemonitor.yaml:12]: I better understand, I didn't get that the selector was for the deployments :)

**Comment by victornoel** [comment]
/approve

(not sure it is needed, but well, at least it's explicit :)

File: stable/pomerium/templates/servicemonitor.yaml (hunk 3 above)
Comment (travisgroth) [reply stable/pomerium/templates/servicemonitor.yaml:12]: 👍

File: stable/pomerium/templates/servicemonitor.yaml (hunk 2 above)
Comment (travisgroth) [reply stable/pomerium/templates/servicemonitor.yaml:7]: 👍

**Comment by travisgroth** [comment]
@victornoel I think i need lgtm as well

**Comment by victornoel** [comment]
Let's try but I don't think I can as a reviewer, you need @desimone.

/lgtm

Le ven. 2 août 2019 à 14:32, Travis Groth <notifications@github.com> a
écrit :

**Comment by victornoel** [comment]
Great, it worked, I was confusing what each role can do :)

**Pull request merged**